openslides/utils/consumers.py

```diff
@@ -100,14 +102,34 @@ class SiteConsumer(ProtocollAsyncJsonWebsocketConsumer):
 
         Sends the startup data to the user.
         """
-        # TODO: add a way to ask for the data since a change_id and send only data that is newer
+        change_id = None
         if not await async_anonymous_is_enabled() and self.scope['user'].id is None:
             await self.close()
-        else:
-            await self.accept()
-            data = await startup_data(self.scope['user'])
+            return
+
+        query_string = parse_qs(self.scope['query_string'])
+        if b'change_id' in query_string:
+            try:
+                change_id = int(query_string[b'change_id'][0])
+            except ValueError:
+                await self.close()  # TODO: Find a way to send an error code
```

Comment:
Maybe we should define our own status codes beginning at 4000 (https://tools.ietf.org/html/rfc6455#section-7.4.2).
like one fpr OK, bad query string, ...